study/test/src/org/labkey/test/tests/study/StudyPublishQueriesTest.java

```diff
@@ -0,0 +1,224 @@
+package org.labkey.test.tests.study;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.labkey.test.BaseWebDriverTest;
+import org.labkey.test.Locator;
+import org.labkey.test.categories.Daily;
+import org.labkey.test.util.DataRegionTable;
+import org.labkey.test.util.StudyHelper;
+import org.openqa.selenium.WebElement;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static org.junit.Assert.assertEquals;
+
+@Category({Daily.class})
+public class StudyPublishQueriesTest extends BaseWebDriverTest
+{
+    public static final String PROJECT_NAME = "StudyPublishQueriesTestProject";
+    public static final String USER_DEFINED_QUERY = "MyUserDefinedQuery";
+    public static final List<String> SCHEMA_NAMES = Arrays.asList("wiki", "core", "study");
+
+    @BeforeClass
+    public static void setupProject()
+    {
+        StudyPublishQueriesTest init = (StudyPublishQueriesTest) getCurrentTest();
+        init.doSetup();
+    }
+
+    @Before
+    public void preTest()
+    {
+        goToProjectHome();
+    }
+
+    private void doSetup()
+    {
+        _containerHelper.createProject(getProjectName(), "Study");
+        _studyHelper.startCreateStudy().createStudy();
+
+        // need at least one visit for the publish study wizard
+        _studyHelper.goToManageVisits().goToImportVisitMap();
+        setFormElement(Locator.name("content"),
+        "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n" +
+                "<visitMap xmlns=\"http://labkey.org/study/xml\">\n" +
+                "  <visit label=\"Test Visit1\" typeCode=\"X\" sequenceNum=\"1.0\" maxSequenceNum=\"1.0\"/>\n" +
+                "  <visit label=\"Test Visit2\" typeCode=\"X\" sequenceNum=\"2.0\" maxSequenceNum=\"2.0\"/>\n" +
+                "  <visit label=\"Test Visit3\" typeCode=\"X\" sequenceNum=\"3.0\" maxSequenceNum=\"3.0\"/>\n" +
+                "</visitMap>"
+        );
```

Comment:
Java 16 text blocks!!
```suggestion
        setFormElement(Locator.name("content"),
        """
        <?xml version=\"1.0\" encoding=\"UTF-8\"?>
          <visitMap xmlns=\"http://labkey.org/study/xml\">
            <visit label=\"Test Visit1\" typeCode=\"X\" sequenceNum=\"1.0\" maxSequenceNum=\"1.0\"/>
            <visit label=\"Test Visit2\" typeCode=\"X\" sequenceNum=\"2.0\" maxSequenceNum=\"2.0\"/>
            <visit label=\"Test Visit3\" typeCode=\"X\" sequenceNum=\"3.0\" maxSequenceNum=\"3.0\"/>
          </visitMap>
        """
        );
```

Reply:
Cool! Done.